### ISSUE TITLE
Add `weight_decay` parameter to NeuralNet

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/deeprank-core#155]
In `NeuralNet` class, the optimizer used during training does not have a `weight_decay` parameter that the user can set, and it defaults to 0 (see [PyTorch documentation](https://pytorch.org/docs/stable/generated/torch.optim.Adam.html)).
Weight decay is fundamental for GNNs, otherwise, parameters can become too big and the gradient may explode.

- The user should be able to edit `weight_decay` and turn it off, so this parameter should be added as input to `NeuralNet`.
- The default for GNNs could be 1e-5.
